pkgs/unified_analytics/lib/src/session.dart

```diff
@@ -87,13 +96,25 @@ class Session {
 
     try {
       parseContents();
-    } on FormatException {
+    } on FormatException catch (err) {
       Initializer.createSessionFile(sessionFile: sessionFile);
 
+      _errorHandler.log(Event.analyticsException(
+        workflow: 'Session._refreshSessionData',
+        error: err.runtimeType.toString(),
+        description: 'message: ${err.message}\nsource: ${err.source}',
+      ));
+
       parseContents();
```

Comment:
Won't this just throw again and not be caught?

Reply:
Yeah this was naively assuming that the error we encounter while parsing the json was user related and not coming from the package, creating a fix for this now